### ISSUE TITLE
Core dump on xml_dump when repo doesn't have module updaterecords

### ISSUE THREAD

**Issue by pavelpicka** [rpm-software-management/createrepo_c#159]
When using python bindings (on latest 0.13.2 createrepo_c) to dump repository metadata (modules available) but no update records (for modules) are present got a core dump.

In case update records for modules are present in the repository (tested with fedora30 repository) everything goes well.

````
systemd-coredump[22564]: Process 22558 (rq) of user 1000 dumped core.

Stack trace of thread 22558:
#0  0x00007face148dffb cr_xml_dump_updatecollectionmodule (_createrepo_c.so)
#1  0x00007face148e123 cr_xml_dump_updateinforecord_pkglist (_createrepo_c.so)
#2  0x00007face148e3fc cr_xml_dump_updateinforecord_internal (_createrepo_c.so)
#3  0x00007face148e453 cr_xml_dump_updateinfo_body (_createrepo_c.so)
#4  0x00007face148e4ab cr_xml_dump_updateinfo (_createrepo_c.so)
#5  0x00007face147d060 xml_dump (_createrepo_c.so)
#6  0x00007face7e22941 _PyMethodDef_RawFastCallKeywords (libpython3.7m.so.1.0)
#7  0x00007face7e4cc0d _PyMethodDescr_FastCallKeywords (libpython3.7m.so.1.0)
#8  0x00007face7e9835d _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#9  0x00007face7e220ca _PyFunction_FastCallKeywords (libpython3.7m.so.1.0)
#10 0x00007face7e98181 _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#11 0x00007face7e4dffb _PyGen_Send (libpython3.7m.so.1.0)
#12 0x00007face7e949c9 _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#13 0x00007face7e4dffb _PyGen_Send (libpython3.7m.so.1.0)
#14 0x00007face5c7a457 n/a (_asyncio.cpython-37m-x86_64-linux-gnu.so)
#15 0x00007face7e4d14c _PyObject_FastCallKeywords (libpython3.7m.so.1.0)
#16 0x00007face7e4d4bf n/a (libpython3.7m.so.1.0)
#17 0x00007face7dfb2ae _PyMethodDef_RawFastCallDict (libpython3.7m.so.1.0)
#18 0x00007face7dfb440 _PyCFunction_FastCallDict (libpython3.7m.so.1.0)
#19 0x00007face7e99656 _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#20 0x00007face7e220ca _PyFunction_FastCallKeywords (libpython3.7m.so.1.0)
#21 0x00007face7e93461 _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#22 0x00007face7e220ca _PyFunction_FastCallKeywords (libpython3.7m.so.1.0)
#23 0x00007face7e93461 _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#24 0x00007face7e220ca _PyFunction_FastCallKeywords (libpython3.7m.so.1.0)
#25 0x00007face7e93461 _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#26 0x00007face7e220ca _PyFunction_FastCallKeywords (libpython3.7m.so.1.0)
#27 0x00007face7e93461 _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#28 0x00007face7e220ca _PyFunction_FastCallKeywords (libpython3.7m.so.1.0)
#29 0x00007face7e93461 _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#30 0x00007face7ddb506 _PyEval_EvalCodeWithName (libpython3.7m.so.1.0)
#31 0x00007face7ddc9e5 _PyFunction_FastCallDict (libpython3.7m.so.1.0)
#32 0x00007face7e94f9e _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#33 0x00007face7e220ca _PyFunction_FastCallKeywords (libpython3.7m.so.1.0)
#34 0x00007face7e93461 _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#35 0x00007face7e220ca _PyFunction_FastCallKeywords (libpython3.7m.so.1.0)
#36 0x00007face7e93461 _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#37 0x00007face7ddb506 _PyEval_EvalCodeWithName (libpython3.7m.so.1.0)
#38 0x00007face7e22271 _PyFunction_FastCallKeywords (libpython3.7m.so.1.0)
#39 0x00007face7e98181 _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#40 0x00007face7ddb506 _PyEval_EvalCodeWithName (libpython3.7m.so.1.0)
#41 0x00007face7e22271 _PyFunction_FastCallKeywords (libpython3.7m.so.1.0)
#42 0x00007face7e93461 _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#43 0x00007face7e220ca _PyFunction_FastCallKeywords (libpython3.7m.so.1.0)
#44 0x00007face7e93461 _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#45 0x00007face7e220ca _PyFunction_FastCallKeywords (libpython3.7m.so.1.0)
#46 0x00007face7e93461 _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#47 0x00007face7ddc5fa _PyFunction_FastCallDict (libpython3.7m.so.1.0)
#48 0x00007face7deb9c6 _PyObject_Call_Prepend (libpython3.7m.so.1.0)
#49 0x00007face7ddcd3b PyObject_Call (libpython3.7m.so.1.0)
#50 0x00007face7e94f9e _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#51 0x00007face7ddb506 _PyEval_EvalCodeWithName (libpython3.7m.so.1.0)
#52 0x00007face7e22271 _PyFunction_FastCallKeywords (libpython3.7m.so.1.0)
#53 0x00007face7e93461 _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#54 0x00007face7ddb506 _PyEval_EvalCodeWithName (libpython3.7m.so.1.0)
#55 0x00007face7e22271 _PyFunction_FastCallKeywords (libpython3.7m.so.1.0)
#56 0x00007face7e944ea _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#57 0x00007face7ddb506 _PyEval_EvalCodeWithName (libpython3.7m.so.1.0)
#58 0x00007face7ddc9e5 _PyFunction_FastCallDict (libpython3.7m.so.1.0)
#59 0x00007face7e94f9e _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#60 0x00007face7ddb506 _PyEval_EvalCodeWithName (libpython3.7m.so.1.0)
#61 0x00007face7ddc9e5 _PyFunction_FastCallDict (libpython3.7m.so.1.0)
#62 0x00007face7deb9c6 _PyObject_Call_Prepend (libpython3.7m.so.1.0)
#63 0x00007face7ddcd3b PyObject_Call (libpython3.7m.so.1.0)

Stack trace of thread 22561:
#0  0x00007face7cb5006 do_futex_wait.constprop.1 (libpthread.so.0)
#1  0x00007face7cb50f8 __new_sem_wait_slow.constprop.0 (libpthread.so.0)
#2  0x00007face7dcd008 PyThread_acquire_lock_timed (libpython3.7m.so.1.0)
#3  0x00007face6f2b8c0 n/a (_queue.cpython-37m-x86_64-linux-gnu.so)
#4  0x00007face7e2295c _PyMethodDef_RawFastCallKeywords (libpython3.7m.so.1.0)
#5  0x00007face7e22ae0 _PyCFunction_FastCallKeywords (libpython3.7m.so.1.0)
#6  0x00007face7e98d1f _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#7  0x00007face7ddc5fa _PyFunction_FastCallDict (libpython3.7m.so.1.0)
#8  0x00007face7e94f9e _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#9  0x00007face7e220ca _PyFunction_FastCallKeywords (libpython3.7m.so.1.0)
#10 0x00007face7e93461 _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#11 0x00007face7e220ca _PyFunction_FastCallKeywords (libpython3.7m.so.1.0)
#12 0x00007face7e93461 _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#13 0x00007face7ddc5fa _PyFunction_FastCallDict (libpython3.7m.so.1.0)
#14 0x00007face7deb9c6 _PyObject_Call_Prepend (libpython3.7m.so.1.0)
#15 0x00007face7ddcd3b PyObject_Call (libpython3.7m.so.1.0)
#16 0x00007face7f04503 n/a (libpython3.7m.so.1.0)
#17 0x00007face7cac58e start_thread (libpthread.so.0)
#18 0x00007face7a4c513 __clone (libc.so.6)

Stack trace of thread 22560:
#0  0x00007face7a43b8f __select (libc.so.6)
#1  0x00007face7e4910f n/a (libpython3.7m.so.1.0)
#2  0x00007face7e22798 _PyMethodDef_RawFastCallKeywords (libpython3.7m.so.1.0)
#3  0x00007face7e22ae0 _PyCFunction_FastCallKeywords (libpython3.7m.so.1.0)
#4  0x00007face7e9880b _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#5  0x00007face7ddb506 _PyEval_EvalCodeWithName (libpython3.7m.so.1.0)
#6  0x00007face7ddc7d7 _PyFunction_FastCallDict (libpython3.7m.so.1.0)
#7  0x00007face7e94f9e _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#8  0x00007face7e220ca _PyFunction_FastCallKeywords (libpython3.7m.so.1.0)
#9  0x00007face7e93461 _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#10 0x00007face7e220ca _PyFunction_FastCallKeywords (libpython3.7m.so.1.0)
#11 0x00007face7e93461 _PyEval_EvalFrameDefault (libpython3.7m.so.1.0)
#12 0x00007face7ddc5fa _PyFunction_FastCallDict (libpython3.7m.so.1.0)
#13 0x00007face7deb9c6 _PyObject_Call_Prepend (libpython3.7m.so.1.0)
#14 0x00007face7ddcd3b PyObject_Call (libpython3.7m.so.1.0)
#15 0x00007face7f04503 n/a (libpython3.7m.so.1.0)
#16 0x00007face7cac58e start_thread (libpthread.so.0)
#17 0x00007face7a4c513 __clone (libc.so.6)
````

**Comment by kontura** [comment]
Thank you for the report.
There is a PR #158  that should fix this problem.

**Comment by Conan-Kudo** [comment]
This should have been fixed with the 0.14.0 release.
